Wizardry/Core/SkillSys/MiscSkillEffects/PrePhaseSkills/ArmorMarch.c

```diff
@@ -0,0 +1,67 @@
+#include "common-chax.h"
+#include "debuff.h"
+#include "kernel-lib.h"
+#include "skill-system.h"
+#include "constants/skills.h"
+
+static void _SetArmorMarchStatDebuff(struct Unit * unit)
+{
+    SetUnitStatDebuff(unit, UNIT_STAT_BUFF_ARMOR_MARCH);
+}
+
+bool PrePhase_TickArmorMarchSkillStatus(ProcPtr proc)
+{
+    int i, j;
+
+    for (i = gPlaySt.faction + 1; i <= (gPlaySt.faction + GetFactionUnitAmount(gPlaySt.faction)); ++i)
+    {
+        bool armor_march  = false;
+
+        struct Unit * unit = GetUnit(i);
+
+        if (!UNIT_IS_VALID(unit))
+            continue;
+
+        if (unit->state & (US_HIDDEN | US_DEAD | US_RESCUED | US_BIT16))
+            continue;
+
+#if (defined(SID_ArmorMarch) && (COMMON_SKILL_VALID(SID_ArmorMarch)))
+        if (SkillTester(unit, SID_ArmorMarch))
+            armor_march = true;
```

Comment:
I think directly "continue" if failed on judgement will be faster?

Reply:
Very good point. No need to start the inner for loop if the current unit doesn't have the skill.